### PR TITLE
Add URI resolver for raster_tindex files

### LINKED ISSUE #856: Create `raster_tindex` driver
### Kind of request

Changing existing functionality

### Enhancement Description

For V1, we need to create a dedicated `raster_tindex` Driver class

### Use case

Read rasters using a tileindex

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Tjalling-dejong**
## Issue addressed
Fixes #856 

## General Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `main`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation
- [x] Updated changelog.rst

## Data/Catalog checklist
- [x] `data/catalogs/predefined_catalogs.yml` has not been modified.
- [x] None of the old `data_catalog.yml` files have been chagned
- [ ] `data/chagnelog.rst` has been updated
- [ ] new file uses `LF` line endings (done automatically if you used `update_versions.py`)
- [ ] New file has been tested locally
- [ ] Tests have been added using the new file in the test suite

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -184,6 +184,14 @@ ConventionResolver
    metadata_resolver.convention_resolver.ConventionResolver
    metadata_resolver.convention_resolver.ConventionResolver.resolve
 
+RasterTindexResolver
+--------------------
+.. autosummary::
+   :toctree: _generated
+
+   metadata_resolver.raster_tindex_resolver.RasterTindexResolver
+      metadata_resolver.raster_tindex_resolver.RasterTindexResolver.resolve
```
Comment (Jaapel) [inline docs/api.rst:193]:
```suggestion
   metadata_resolver.raster_tindex_resolver.RasterTindexResolver
   metadata_resolver.raster_tindex_resolver.RasterTindexResolver.resolve
```

Comment (savente93) [inline docs/api.rst:193]:
```suggestion
   metadata_resolver.raster_tindex_resolver.RasterTindexResolver.resolve
```

**Review by savente93 — COMMENTED** [review]
nice work, tests look good, just one white space typo. I'll leave Jaap to do the final approve, since he's better versed in this part of the code

File: hydromt/metadata_resolver/raster_tindex_resolver.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        zoom_level: Optional[ZoomLevel] = None,
+        variables: Union[int, tuple[float, str], None] = None,
+        handle_nodata: NoDataStrategy = NoDataStrategy.RAISE,
+        logger: Optional[Logger] = logger,
+        options: Optional[Dict[str, Any]] = None,
+    ) -> List[str]:
+        """Resolve URIs of a raster tindex file."""
+        gdf = gpd.read_file(uri)
+        gdf = gdf.iloc[gdf.sindex.query(mask.to_crs(gdf.crs).unary_union)]
+        tileindex = options.get("tileindex")
```
Comment (Jaapel) [inline hydromt/metadata_resolver/raster_tindex_resolver.py:36]:
```suggestion
        tileindex: Optional[str] = options.get("tileindex")
        if tileindex is None:
            raise ValueError(f"{self.__class__.__name__} needs options specifying 'tileindex'}")
```

**Review by Jaapel — APPROVED** [review]
Thanks @Tjalling-dejong!
Looks good, a few minor suggestions.

**Pull request merged**